llvm/lib/Target/EraVM/EraVMISelLowering.cpp

```diff
@@ -1029,6 +1045,63 @@ SDValue EraVMTargetLowering::LowerBR_JT(SDValue Op, SelectionDAG &DAG) const {
   return SDValue(Res, 0);
 }
 
+// If the condition of SELECT instruction comes from a overflow intrinsic, lower
+// it into conditional move.  A simple case is as below:
+//
+//   %res1 = call {i256, i1} @llvm.uadd.with.overflow.i256(i256 %x, i256 %y)
+//   %overflow = extractvalue {i256, i1} %res1, 1
+//   %selected = select i1 %overflow, i256 %a, i256 %b
+//
+// which can be lowered to
+//
+//   add!	  %x, %y, r3
+//   add    %b, r0, %selected
+//   add.of	%a, r0, %selected
+//
+// Please be noted that it is incorrect to use GE as negation
+// code for overflow LT.
+SDValue EraVMTargetLowering::LowerSELECT(SDValue Op, SelectionDAG &DAG) const {
+  SDValue Cond = Op.getOperand(0);
+  SDValue TrueV = Op.getOperand(1);
+  SDValue FalseV = Op.getOperand(2);
+
+  SDValue MatchedUArithO = matchingOverflowArithmeticOperation(Cond);
+  if (!MatchedUArithO)
+    return SDValue();
+
+  SDValue FoldedArith;
+  auto Opc = MatchedUArithO.getOpcode();
+  auto LoweredOpc = OpcodeMap.at(Opc);
+  SDLoc DL(Op);
+  SDVTList FoldedVT = DAG.getVTList(MVT::i256, MVT::Other, MVT::Glue);
+  int OFGlueResult = 2;
+  // MUL has 2 results
+  if (LoweredOpc == EraVMISD::MUL_V) {
+    FoldedVT = DAG.getVTList(MVT::i256, MVT::i256, MVT::Other, MVT::Glue);
+    OFGlueResult = 3;
+  }
+  FoldedArith = DAG.getNode(LoweredOpc, DL, FoldedVT,
+                            {DAG.getEntryNode(), MatchedUArithO.getOperand(0),
+                             MatchedUArithO.getOperand(1)});
+  // Deliberately comment out this call in the original patch to support below
```

Comment:
I think it's better to avoid referencing the original patch as not everybody should know history of our project.
I suggest to put it this way:
Note that we create a new node while keeping the original one. This is because of the following case.
<example>
We need to keep...

Reply:
Got it and will reword.